### ISSUE TITLE
ERR! on npm run storybook

### ISSUE THREAD

**Issue by Strutsagget** [kadirahq/react-storybook#145]
I get thesse errors: What am i doing wrong?
npm run storybook
npm ERR! Darwin 15.4.0
npm ERR! argv "/usr/local/Cellar/node/5.10.1/bin/node" "/usr/local/bin/npm" "run" "storybook"
npm ERR! node v5.10.1
npm ERR! npm  v3.8.3

npm ERR! missing script: storybook

**Comment by mnmtanish** [comment]
I think you forgot to add the npm script. Add `"storybook": "start-storybook -p 9001"` under `scripts` in your package.json file. [more info](https://github.com/kadirahq/react-storybook#getting-started)

**Comment by otmjka** [comment]
The same situation:
`
"scripts": {
 "storybook": "start-storybook -p 9001"
}
`

[npm-debug.log.txt](https://github.com/kadirahq/react-storybook/files/239072/npm-debug.log.txt)

**Comment by decoursin** [comment]
I'm having the same issue. The storybook script is there, but I'm getting: "sh: 1: start-storybook: not found"

**Comment by arunoda** [comment]
@decoursin Are you using windows?

**Comment by decoursin** [comment]
I'm using Linux.

**Comment by decoursin** [comment]
I switched from npm 2.x.z to 3.y,z, rm -rf node_modules, and now it's working. I'm not sure at all if this was the issue.

**Comment by otmjka** [comment]
Switched the storybook to 1.6.0 version, and it started working fine. I'm using react 0.14.8.

**Comment by otmjka** [comment]
created stories + config. 
`npm  run storybook`
after updated to 1.10.0 after to last, in both cases the storybook worked.

**Comment by jurgob** [comment]
I had the same error becouse I had installed npm install react-storybook instead of npm install @kadira/storybook

**Comment by szopenkrk** [comment]
Hi Guys i have same problem $ npm run storybook
npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "storybook"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.4

npm ERR! missing script: storybook

**Comment by arunoda** [comment]
Hmm. Seems like we are only support bash. We need to add support for windows.

**Comment by mnmtanish** [comment]
@szopenkrk can you follow these steps and try again?

- add the npm script `storybook` to your projects package.json file if it's missing

```diff
{
  "scripts": {
+    "storybook": "start-storybook -p 9001"
  }
}
```

- install npm3 [(more info)](https://github.com/kadirahq/react-storybook/blob/master/docs/known_issues.md#potential-issues-with-npm-2)

```
npm i -g npm@3
rd /s /q node_modules
npm i
```

if you're trying to use storybook with your react app, install from npm

```
npm i -D @kadira/storybook
```

If you're trying to contribute to this repo (thanks) please wait a few days until we make the prepublish script cross platform ( or you can help us out with that issue itself #300 ).

**Comment by szopenkrk** [comment]
I made that but on Windows still i have error
`karol@FIZZ MINGW64 /c/Dev/react-storybook-demo (master)
$ npm run storybook

> redux-todomvc-example@0.0.0 storybook C:\Dev\react-storybook-demo
> start-storybook -p 9001

'start-storybook' is not recognized as an internal or external command,
operable program or batch file.

npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "storybook"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.4
npm ERR! code ELIFECYCLE
npm ERR! redux-todomvc-example@0.0.0 storybook: `start-storybook -p 9001`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the redux-todomvc-example@0.0.0 storybook script 'start-storybook -p 9001'.
npm ERR! This is most likely a problem with the redux-todomvc-example package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     start-storybook -p 9001
npm ERR! You can get their info via:
npm ERR!     npm owner ls redux-todomvc-example
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Dev\react-storybook-demo\npm-debug.log
`

**Comment by szopenkrk** [comment]
OK i have solution 

```
npm i -g npm@3.8.5
rm -rf node_modules
npm i
```

and in demo we dont need add "storybook": "start-storybook -p 9001" because we have it there.

**Comment by satstanford** [comment]
I am also encountering this problem, despite trying all the fixes suggested above (including removing node modules and reinstalling them, adding storybook to the scripts in package.json etc)
I am using npm3,TypeScript 1.8.10, I installed with "npm i -D @kadira/storybook", and I am using Git Bash on a Windows 10 computer. 
I attached my npm-debug log

[npm-debug.txt](https://github.com/kadirahq/react-storybook/files/389671/npm-debug.txt)

**Comment by arunoda** [comment]
@satstanford try to use `start-storybook` instead `./node_modules/.bin/start-storybook`

**Comment by jonathanconway** [comment]
@mnmtanish thanks for your comment on installing from `npm`.

Could we add this command to the docs...

```
npm i -D @kadira/storybook
```

...explaining that this is how to install if you to run `storybook-start` from your npm scripts without installing Storybook globally?

**Comment by arunoda** [comment]
@jonathanconway I think everything is documented properly.
See: https://getstorybook.io/docs/react-storybook/basics/slow-start-guide

**Comment by danideng** [comment]
This is package.json created after running create-react-app to create my-stories-app. You need to pay attentions to the default versions created for "react", "react-dom" and "react-scripts" under "dependencies":
`{
  "name": "my-stories-book",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "react": "^16.0.0",
    "react-dom": "^16.0.0",
    "react-scripts": "1.0.14"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "devDependencies": {
    "@kadira/storybook": "^2.35.3"
  }
}
`

I probably looked through every answer, tried them, all not working. I even had gone to the point that finally "npm run storybook" didn't give me any error after "npm i -D @kadira/storybook" as provided by mnmtanish, but when I access http://localhost:6006, there was a run-time JS error that stated "... string ... not found" which I didn't know what it meant and didn't bother to trace.

Here is my new package.json in which I used specific versions that finally could bring up the app problems free. Again, pay attention to the versions of "react", "react-dom" and "react-scripts" under "dependencies".
`{
  "name": "my-stories-book",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "react": "^15.4.2",
    "react-dom": "^15.4.2",
    "react-scripts": "0.8.5"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "devDependencies": {
    "@kadira/storybook": "^2.35.3"
  }
}
`

The above is the proof that getstorybook is not updated with the default versions created by create-react-app.

**Comment by Hypnosphi** [comment]
@danideng Looks like you're using the outdated `getstorybook` package, the up-to-date one is called `@cli/storybook`

Related: #1604

**Comment by danideng** [comment]
Based on the information provided by Hypnosphi, here are the steps of my second try:

By following https://www.npmjs.com/package/@storybook/cli:
1. C:\...\test > create-react-app cli-story-book
2. C:\...\test > npm i -g @storybook/cli
3. C:\...\test > cd cli-story-book
4. C:\...\test > getstorybook
 • Detecting project type. ✓
 • Adding storybook support to your "Create React App" based project. ✓
 • Preparing to install dependencies. ✓
......
5. C:\...\test > npm run storybook
......
Storybook started on => http://localhost:9009/
===============================

Voila! Thank you very much, Hypnosphi.

**Comment by Sebs030** [comment]
@3DEsprit  Thanks, just ran into the same issue, that `npx -p @storybook/cli sb init` fixed the issue

**Comment by rodrigouz** [comment]
> Based on the information provided by Hypnosphi, here are the steps of my second try:
> 
> By following https://www.npmjs.com/package/@storybook/cli:
> 
> 1. C:...\test > `create-react-app cli-story-book`
> 2. C:...\test > `npm i -g @storybook/cli`
> 3. C:...\test > `cd cli-story-book`
> 4. C:...\test > `getstorybook`
>    • Detecting project type. ✓
>    • Adding storybook support to your "Create React App" based project. ✓
>    • Preparing to install dependencies. ✓
>    ......
> 5. C:...\test > `npm run storybook`
>    ......
>    Storybook started on => http://localhost:9009/
>    ===============================
> 
> Voila! Thank you very much, Hypnosphi.

in the step 4 I added `init`
>4. >`getstorybook init`

**Comment by gpresland** [comment]
I ran into this issue when I copied my `.storybook` folder from an existing repo to a new one, before installing Storybook with `npx -p @storybook/cli sb init`. If you run `npx -p @storybook/cli sb init` and already contain a `.storybook` folder, it seems Storybook will skip some setup.

The solution was to simply delete my `.storybook` folder and run `npx -p @storybook/cli sb init -f` (with a `-f` to force reinstall).
